### DISCUSSION TITLE
Proof of Concept Twitter Alternate UI task

### DISCUSSION THREAD

**Discussion by dennyabrain** [tattle-made/Uli#29] [General]
# Task
Create an alternative frontend for Twitter. Take a look at [nitter](https://nitter.net/about) as an inspiration.
	
# Deliverable
A webapp running on localhost. User should be able to enter any twiter user's name in a text field and see their feed in the webapp.
A key requirement is to not use some third party library for getting tweets from twitter but to use some web scraping library for the same.

I know of the following 2 that are popular in the JS ecosystem
https://github.com/cheeriojs/cheerio
https://github.com/puppeteer/puppeteer

**Comment by dennyabrain** [comment]
You are free to use Ruby/Python for your backend server for this task if you are more comfortable with that.

**Comment by newhorizon-tech** [comment]
- I was doing some research and weighing the options in terms of choosing the right tech. I think I'll go with Node.js as backend/React as front-end. I am more comfortable with Ruby but I think this could also be a good opportunity to practice Node. I'll use Puppeteer for the web-scraping part.  (https://github.com/newhorizon-tech/fresh-feed/issues/1)

- I'll try to make some progress on creating a basic front-end, then try to make the scraper work and then I'll refine the frontend at the end  

- I have also created a kanban board where I'll add issues and keep track of mini-tasks. (https://github.com/newhorizon-tech/fresh-feed/projects/1). I'll also post major updates on the current discussion page.

**Reply by dennyabrain** [reply]
Oh great. I'll follow the linked kanban and issue.

Great idea on point 2. I think the scraping bit will definitely be more involved and I've seen your frontend chops in your other portfolio projects, so having a minimal/basic UI will be just fine. So it ok to allot more time to scraping and figuring out what kind of data we can get.
